servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletRegistration.java

```diff
@@ -0,0 +1,252 @@
+     */
+    public ServletRegistration(String servletName, Servlet servlet, DefaultServletContext servletContext,
+                               UrlMapper<ServletRegistration> urlMapper, Map<String, String> initParameterMap) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(servlet, "servlet");
+        requireNonNull(servletContext, "servletContext");
+        requireNonNull(urlMapper, "urlMapper");
+
+        this.servletName = servletName;
+        this.servlet = servlet;
+        this.servletContext = servletContext;
+        this.urlMapper = urlMapper;
+        this.initParameterMap = ImmutableMap.copyOf(initParameterMap);
+        servletConfig = new ServletConfig() {
+            @Override
+            public String getServletName() {
+                return ServletRegistration.this.servletName;
+            }
+
+            @Override
+            public ServletContext getServletContext() {
+                return ServletRegistration.this.servletContext;
+            }
+
+            @Override
+            @Nullable
+            public String getInitParameter(String name) {
+                return ServletRegistration.this.getInitParameter(name);
+            }
+
+            @Override
+            public Enumeration<String> getInitParameterNames() {
+                return Collections.enumeration(ImmutableSet.copyOf(getInitParameters().keySet()));
+            }
+        };
+    }
+
+    /**
+     * Get servlet security element.
+     */
+    @Nullable
+    public ServletSecurityElement getServletSecurityElement() {
+        return servletSecurityElement;
+    }
+
+    /**
+     * Get multipart config element.
+     */
+    @Nullable
+    public MultipartConfigElement getMultipartConfigElement() {
+        return multipartConfigElement;
+    }
+
+    /**
+     * Get servlet config.
+     */
+    public ServletConfig getServletConfig() {
+        return servletConfig;
+    }
+
+    /**
+     * Get servlet.
+     */
+    public Servlet getServlet() {
+        return servlet;
+    }
+
+    /**
+     * Is async supported.
+     */
+    public Boolean isAsyncSupported() {
+        return asyncSupported;
+    }
+
+    /**
+     * Get load on startup.
+     */
+    public int getLoadOnStartup() {
+        return loadOnStartup;
+    }
+
+    /**
+     * Is initialization servlet cas.
+     */
+    public boolean isInitServletCas(boolean expect, boolean update) {
+        return initServlet.compareAndSet(expect, update);
+    }
+
+    /**
+     * Is initialization servlet.
+     */
+    public boolean isInitServlet() {
+        return initServlet.get();
+    }
+
+    @Override
+    public Set<String> addMapping(String... urlPatterns) {
+        requireNonNull(urlPatterns, "urlPatterns");
+        mappingSet.addAll(Arrays.asList(urlPatterns));
+        for (String pattern : urlPatterns) {
+            urlMapper.addMapping(pattern, this, servletName);
+        }
+        return mappingSet;
```

Comment:
It seems like we need to return the `mappingSet` that `urlPatterns` are added before.

Reply:
yes, each servlet have one `ServletRegistration` to `addMapping` multiple `urlPatterns` to one `servletName` such as: urlPattern (`"/home", "/home/user", "/home/*"`) map to servletName (`"home"`). 
I think, mappingSet is returned to check which urlPatterns were mapped.